internal/ignition/ignition.go

```diff
@@ -45,21 +46,23 @@ type Generator interface {
 }
 
 type installerGenerator struct {
-	log          logrus.FieldLogger
-	workDir      string
-	cluster      *common.Cluster
-	releaseImage string
-	installerDir string
+	log           logrus.FieldLogger
+	workDir       string
+	cluster       *common.Cluster
+	releaseImage  string
+	installerDir  string
+	serviceCACert string
 }
 
 // NewGenerator returns a generator that can generate ignition files
-func NewGenerator(workDir string, installerDir string, cluster *common.Cluster, releaseImage string, log logrus.FieldLogger) Generator {
+func NewGenerator(workDir string, installerDir string, cluster *common.Cluster, releaseImage string, serviceCACert string, log logrus.FieldLogger) Generator {
```

Comment:
I feel like this is going to quickly get out of hand.

I'm not sure a parameter that is only used in some cases belongs in the generator directly. It might be worth refactoring this a bit so the user of the generator would call separate methods to make optional edits to the ignitions like adding the ca cert.

I would imagine something like this:
```
g := ignition.NewGenerator(workDir, installerDir, cluster, releaseImage, log)
g.Generate(installConfig)

if <some_condition> {
  g.AddCaCerts(serviceCACert)
}
if <some_other_condition> {
  g.SomeOtherChange(...)
}

g.UploadToS3(...)
```

Reply:
I see what you mean, but keep in mind that we have two implementations where the conditional logic needs to be done: job and local_job. I am also curious what it would mean with the new hostname (host level) and etc_hosts (cluster wide) ignition changes coming soon. I think we should keep it as is right now, and reevaluate with those new changes.

Reply:
> but keep in mind that we have two implementations where the conditional logic needs to be done: job and local_job

Actually that's a really good point. Won't this only ever need to be used for the on-prem case? (i.e. local job only?)

Also, obviously not for this PR, but now that this isn't actually using a kube job should we decouple this from the job construct at some point? @mhrivnak did you try this at all when moving from the kube job to local? What were the blockers (other than touching a bunch of code that we didn't want to at the time)?